ninshiki_yolo/detector/node/detection.py

```diff
@@ -0,0 +1,127 @@
+# Copyright (c) 2021 Ichiro ITS
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+import cv2
+import numpy as np
+import os
+from rclpy.node import MsgType
+from ninshiki_interfaces.msg import DetectedObject, DetectedObjects
+from shisen_interfaces.msg import Image
+from ninshiki_yolo.utils.draw_detection_result import draw_detection_result
+
+class Detection:
+    def __init__(self, 
+                 config: str = os.path.expanduser('~') + "/yolo_model/config.cfg", 
+                 names: str = os.path.expanduser('~') + "/yolo_model/obj.names",
+                 weights: str = os.path.expanduser('~')+ "/yolo_model/yolo_weights.weights",
+                 gpu: bool = False, 
+                 myriad: bool = False):
+        self.file_name = names
+        self.classes = None
+
+        self.net = cv2.dnn.readNetFromDarknet(config, weights)
+        self.outs = None
+
+        self.width = 0
+        self.height = 0
+
+        self.gpu = gpu
+        self.myriad = myriad
+
+    def pass_image_to_network(self, image: np.ndarray):
+        with open(self.file_name, 'rt') as f:
+            self.classes = f.read().rstrip('\n').split('\n')
+        
+        self.net.getLayerNames()
+        layerOutput = self.net.getUnconnectedOutLayersNames()
+
+        if self.gpu:
+            self.net.setPreferableBackend(cv2.dnn.DNN_BACKEND_CUDA)
+            self.net.setPreferableTarget(cv2.dnn.DNN_TARGET_CUDA)
+        elif self.myriad:
+            self.net.setPreferableBackend(cv2.dnn.DNN_BACKEND_OPENCV)
+            self.net.setPreferableTarget(cv2.dnn.DNN_TARGET_MYRIAD)
+        else:
+            self.net.setPreferableBackend(cv2.dnn.DNN_BACKEND_OPENCV)
+            self.net.setPreferableTarget(cv2.dnn.DNN_TARGET_CPU)
+
+        inpWidth = 416
```

Comment:
There were still a naming convention inconsistency. Maybe you can check all class and see whether its members are using the appropriate naming convention or not.

Reply:
done